flang/test/Lower/OpenMP/omp-parallel-allocate-clause.f90

```diff
@@ -0,0 +1,29 @@
+! This test checks lowering of OpenMP parallel Directive with
+! `PRIVATE` clause present.
+
+! RUN: %bbc -fopenmp -emit-fir %s -o  - | \
+! RUN:   FileCheck %s --check-prefix=FIRDialect
+
+!FIRDialect: func @_QPallocate_clause(%arg0: !fir.ref<f32>, %arg1: !fir.ref<f32>) {
+!FIRDialect-DAG: %[[A:.*]] = fir.alloca i32 {name = "{{.*}}Ea"}
+!FIRDialect-DAG: %[[B:.*]] = fir.alloca i32 {name = "{{.*}}Eb"}
+!FIRDialect-DAG: %[[X:.*]] = fir.alloca i32 {name = "{{.*}}Ex"}
+!FIRDialect-DAG: %[[Y:.*]] = fir.alloca i32 {name = "{{.*}}Ey"}
+!FIRDialect-DAG: %[[Z:.*]] = fir.alloca i32 {name = "{{.*}}Ez"}
+
+!FIRDialect-DAG: %[[LARGE:.*]] = constant 1 : i32
+!FIRDialect-DAG: %[[DEFAULT:.*]] = constant 1 : i32
```

Comment:
Why are these the same for both default and large mem allocation? Shouldn't they be different constants?

Reply:
I'm not entirely sure as the constants are taken from the parse tree, except in the "no allocator specified" case, where I pass 1 as the constant.

Reply:
Ahh OK. I found that the entry of large_cap_mem_alloc is set to default in the omp_lib.h file. We have to either update this file or use the one provided with the OpenMP runtime in flang.

1) Relevant info in Flang omp_lib.f90 (it includes omp_lib.h)
https://github.com/llvm/llvm-project/blob/13042da5cb3fe5f17c58c2446c93635a2c432049/flang/module/omp_lib.h#L106

2) Relevant info in OpenMP runtime omp_lib.f90
https://github.com/llvm/llvm-project/blob/13042da5cb3fe5f17c58c2446c93635a2c432049/openmp/runtime/src/include/omp_lib.f90.var#L132